File: sgifd-backend/src/main/java/com/sgifdbackend/demo/parametrage/entites/Commune.java
package com.sgifdbackend.demo.parametrage.entites;

import com.sgifdbackend.demo.entitiesBase.EntityBaseBean;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "t_Commune")
public class Commune extends EntityBaseBean implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String libelle;

    @ManyToOne
    private Departement departement;

    public Commune() {
    }

    public Departement getDepartement() {
		return departement;
	}

	public void setDepartement(Departement departement) {
		this.departement = departement;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getLibelle() {
        return libelle;
    }

    public void setLibelle(String libelle) {
        this.libelle = libelle;
    }


}

File: sgifd-frontend/src/app/models/Commune.ts
import {Departement} from "./Departement";

export class Commune {
  id: number;
  libelle:	string;
  createBy:	string;
  deleteBy:	string;
  departement: Departement;

  constructor(id: number, libelle: string, createBy: string, deleteBy: string, departement: Departement) {
    this.id = id;
    this.libelle = libelle;
    this.createBy = createBy;
    this.deleteBy = deleteBy;
    this.departement = departement;
  }
}
